### ISSUE TITLE
Critical Path Tree View

### ISSUE THREAD

**Issue by syamajala** [StanfordLegion/prof-viewer#70]
It would be nice if we had a tree view for the critical path to make it easier/more efficient to navigate.

The way I sort of imagine it is that when you click critical path for an item, it opens a scrollable pane with a tree that shows you both the upstream and downstream things that you are waiting on/waiting for you and when you hover over an item in the tree it will pan/zoom to that item in the profile and there would be a pane that shows you that items info instead of popping up another window.

**Comment by syamajala** [comment]
Seems like its possible with egui: https://github.com/blitzarx1/egui_graphs

**Comment by elliottslaughter** [comment]
Someone will need to help me upgrade egui if we're going to use this; they're on the newest version and we're still on 0.25.

**Comment by lightsighter** [comment]
What would be required to upgrade to the latest egui? Does it have backwards breaking changes?

**Comment by elliottslaughter** [comment]
egui 0.26 introduces text selection, which is great but breaks all our custom widgets. Because they rearchitected the way that inputs are handled, I haven't been able to trace through all the new abstractions to figure out how to do what we need to do yet.

**Comment by elliottslaughter** [comment]
Splitting this discussion of into its own issue: https://github.com/StanfordLegion/prof-viewer/issues/71
